src/utils.jl

```diff
@@ -228,28 +260,35 @@ whatever units), you can specify that behavior by supplying
 
 Handling of units in the conversion process is ignored in this version.
 """
-function extensive_kinetic_math(
+extensive_kinetic_math(
     m::SBML.Model,
     formula::SBML.Math;
     handle_empty_compartment_size = (id::String) -> throw(
         DomainError(
             "Non-substance-only-unit reference to species `$id' in an unsized compartment `$(m.species[id].compartment)'",
         ),
     ),
-)
-    conv(x::SBML.MathIdent) = begin
+) = interpret_math(
+    formula,
+    map_apply = (x, rec) -> SBML.MathApply(x.fn, rec.(x.args)),
+    map_const = identity,
+    map_ident = (x::SBML.MathIdent) -> begin
         haskey(m.species, x.id) || return x
         sp = m.species[x.id]
         sp.only_substance_units && return x
         sz = m.compartments[sp.compartment].size
-        isnothing(sz) && (sz = handle_empty_compartment_size(x.id))
-        SBML.MathApply("/", [x, SBML.MathVal(sz)])
-    end
-    conv(x::SBML.MathApply) = SBML.MathApply(x.fn, conv.(x.args))
-    conv(x::SBML.Math) = x
-
-    conv(formula)
-end
+        isnothing(sz) &&
+            !isboundbyrules(sp.compartment, m) &&
+            (sz = handle_empty_compartment_size(x.id))
```

Comment:
In a case where the size is nothing and the compartment is not bound by rule we end up throwing an error here. However, before we do that, we should check whether `c.spatial_dimensions == 0`, in which case we set `sz = 1.` and don't need to throw an error, right?